types/index.d.ts

```diff
@@ -289,10 +289,61 @@ declare namespace PhonegapPluginPush {
 		from?: string
 		notId?: string
 	}
-
+  /**
+   * Set Channel Variable,once created cannot be changed programmtically for importance and vibration
+   */
+  interface channel{
+    /**
+     * channel ID
+     */
+    id: string
+    /**
+     * camillebeaumont name and description fix
+     */
+    name: string 
+    /**
+     * camillebeaumont name and description fix,channel description
+     */
+    description: string
+    /**
+     * channel importance level
+     */
+    importance: integer
+    /**
+     * enable vibration for the notification
+     */
+    vibration: boolean
+    /**
+     * Set notitfication visibility 
+     */
+    visibility: integer
+    /**
+     * Set Channel Group
+     */
+    groupId:string
+  }
+  
+   interface channelGroup{
+    /**
+     * Set Channel Group ID
+     */
+    id: string
+   /**
+     * Set Channel Group Name
+     */
+    name: string
+  }
+  
 	interface PushNotificationStatic {
 		init(options: InitOptions): PushNotification
 		new (options: InitOptions): PushNotification
+    hasPermission(successHandler: () => any): void 
+    createChannel(successHandler: () => any, errorHandler: () => any,channel): void
+    listChannels(successHandler: () => any): void
+    deleteChannel(successHandler: () => any, errorHandler: () => any, channelId: string ): void
+    createChannelGroup(successHandler: () => any, errorHandler: () => any,channelGroup): void
```

Comment:
The `channelGroup` parameter should be explicitly typed as the interface that represents it.
```suggestion
    createChannelGroup(successHandler: () => any, errorHandler: () => any, channelGroup: ChannelGroup): void
```